journal/src/test/java/io/zeebe/journal/file/JournalReaderTest.java

```diff
@@ -171,6 +171,43 @@ void shouldSeekToHighestAsqnLowerThanProvidedAsqn() {
     assertThat(reader.next()).isEqualTo(expectedRecord);
   }
 
+  @Test
+  void shouldSeekToHighestAsqnWithinBoundIndex() {
+    // given
+    final var firstIndex = journal.append(1, data).index();
+    final var secondIndex = journal.append(4, data).index();
+    final var thirdIndex = journal.append(data).index();
+    final var fourthIndex = journal.append(5, data).index();
+    final var fifthIndex = journal.append(data).index();
```

Comment:
```suggestion
    journal.append(data).index();
```
fifthIndex is never used since you use `Long.MAX_VALUE` for the last seek